servers/visual_server.cpp

```diff
@@ -848,13 +848,15 @@ Error VisualServer::_surface_set_data(Array p_arrays, uint32_t p_format, uint32_
 }
 
 uint32_t VisualServer::mesh_surface_get_format_offset(uint32_t p_format, int p_vertex_len, int p_index_len, int p_array_index) const {
+	ERR_FAIL_INDEX_V(p_array_index, ARRAY_MAX, ERR_PARAMETER_RANGE_ERROR);
```

Comment:
These functions don't return the ``Error`` type, so you shouldn't be returning an error. Instead, you should return ``0``

Reply:
Just to make sure - I should change these to `ERR_FAIL_INDEX` and then have the `mesh_surface` functions return `0` at that point? Or maybe `MSG` to clarify that it will be returning `0`?

Reply:
```suggestion
	ERR_FAIL_INDEX_V(p_array_index, ARRAY_MAX, 0);
```
I would just do this

Reply:
Oh, lol, that's way easier. For some reason I had it in my head it had to be an error value for `ERR_FAIL_INDEX_V`.

Reply:
Haha yep, much easier! The return value should be something that makes sense with the return value of the function and allows the engine to keep on running (or fail gracefully if need be). For most functions, that means returning the default value for the return type (e.g. ``0`` for ``ints`` or ``Vector2()`` for ``Vector2``s) in this case ``0`` is fine!

Reply:
Yeah, that makes too much sense, haha. I appreciate the primer! I just rebased to a single commit as well, as I forgot to amend.